actions/service.go

```diff
@@ -174,66 +183,101 @@ func (s *Service) runTasks(ctx context.Context, record graveler.HookRecord, task
 	return g.Wait().ErrorOrNil()
 }
 
-func (s *Service) insertRunInformation(ctx context.Context, record graveler.HookRecord, tasks [][]*Task) error {
+func (s *Service) saveRunInformation(ctx context.Context, record graveler.HookRecord, tasks [][]*Task) error {
 	if len(tasks) == 0 {
 		return nil
 	}
-	// collect run information based on tasks
-	runStartTime, runEndTime, runPassed := runInformationBasedOnTasks(tasks)
 
-	// save run and tasks information
+	manifest := buildRunManifest(record, tasks)
+
+	err := s.insertRunInformation(ctx, record.RepositoryID, manifest)
+	if err != nil {
+		return fmt.Errorf("insert run information: %w", err)
+	}
+
+	return s.saveRunManifest(ctx, record, manifest)
+}
+
+func (s *Service) saveRunManifest(ctx context.Context, record graveler.HookRecord, manifest runManifest) error {
+	manifestJSON, err := json.Marshal(manifest)
+	if err != nil {
+		return fmt.Errorf("marshal run manifest: %w", err)
+	}
+	runManifestPath := FormatRunManifestOutputPath(record.RunID)
+	manifestReader := bytes.NewReader(manifestJSON)
+	manifestSize := int64(len(manifestJSON))
+	storageNamespace := record.StorageNamespace.String()
+	return s.Writer.OutputWrite(ctx, storageNamespace, runManifestPath, manifestReader, manifestSize)
+}
+
+func (s *Service) insertRunInformation(ctx context.Context, repositoryID graveler.RepositoryID, manifest runManifest) error {
 	_, err := s.DB.Transact(func(tx db.Tx) (interface{}, error) {
 		// insert run information
+		run := manifest.Run
 		_, err := tx.Exec(`INSERT INTO actions_runs(repository_id, run_id, event_type, start_time, end_time, branch_id, source_ref, commit_id, passed)
 			VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9)`,
-			record.RepositoryID, record.RunID, record.EventType, runStartTime, runEndTime, record.BranchID, record.SourceRef, record.CommitID, runPassed)
+			repositoryID, run.RunID, run.EventType, run.StartTime, run.EndTime, run.BranchID, run.SourceRef, run.CommitID, run.Passed)
```

Comment:
`run.CommitID` is always empty in this stage, no?

Reply:
yes, but when you call Run for post events - it will include the commit ID